CMakeLists.txt

```diff
@@ -8,8 +8,8 @@ project(gssw)
 set(CMAKE_CXX_STANDARD 14)
 
 # Use all standard-compliant optimizations
-set (CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -O3 -g -msse4")
```

Comment:
@mr-c If we take out -msse4, what target machine is it going to default to using on amd64? I know that we need to make sure that for vg binary releases and Dockers we *don't* use AVX512 instructions in vg or any of its dependencies, because they're not widely deployed enough in our world yet, and one way we have been avoiding that is liberally sprinkling specific pre-AVX512 `-m` flags throughout the build system. Just letting GCC always build for the current CPU is insufficient; we need to target a lowest common denominator by default.

Maybe with your other cflag propagation changes we can control architecture centrally in the vg Makefile?

Reply:
[Debian/Ubuntu gcc defaults](https://wiki.debian.org/ArchitectureSpecificsMemo#Architecture_baselines) to `-msse2` on 64 bit AMD/Intel with no `-march` target nor -`mtune` target:

Without `-march={native,knl,knm,skylake-avx512,cannonlake,icelake-client,icelake-server,cascadelake}` or one of the `-mavx512` flags, there will be no AVX512 instructions.

FYI: `-msse4` does NOT disable `-mavx512` if that was set elsewhere. Those `-m` flags are cumulative.

If you were really paranoid you could add `-mno-avx512f -mno-avx512pf -mno-avx512er -mno-avx512cd -mno-avx512vl -mno-avx512bw -mno-avx512dq -mno-avx512ifma -mno-avx512vbmi -mno-avx512vbmi2 -mno-avx512bitalg -mno-avx512vpopcntdq -mno-avx5124fmaps -mno-avx512vnni -mno-avx5124vnniw` when on x86 but some of those flags might not be supported by certain gcc versions and they do NOT work on non-x86 gcc.